l1-contracts/src/core/libraries/HeaderLib.sol

```diff
@@ -110,10 +110,11 @@ library HeaderLib {
 
   /**
    * @notice Decodes the header
+   * @dev Marked as public to reduce Rollup.sol contract size
    * @param _header - The header calldata
    * @return The decoded header
    */
-  function decode(bytes calldata _header) internal pure returns (Header memory) {
+  function decode(bytes calldata _header) public pure returns (Header memory) {
```

Comment:
Did some pretty nasty things to bring down the contract size in this commit (64586d8b8760157d2821045557af5a2111ed9552), will look over again once tests are passing.